packages/gatsby-transformer-sharp/src/gatsby-node.js

```diff
@@ -24,7 +24,7 @@ exports.sourceNodes = ({ actions }) => {
 
   if (createTypes) {
     createTypes(`
-      type ImageSharp implements Node @infer {
+      type ImageSharp implements Node @infer @childOf(types: ["File"]) {
```

Comment:
What will happen when version of gatsby doesn't handle those newer directives? Will this be silent no-op or will it complain about unknown directives?

Reply:
Only directives handled here are turned into extensions, so this is a noop in older versions:
https://github.com/gatsbyjs/gatsby/blob/42317c16ea1f86b1b3fe04b91cc25e16b79f757b/packages/gatsby/src/schema/schema.js#L322-L355